src/components/form/Calendar/index.tsx

```diff
@@ -8,52 +8,84 @@ import { useDisplay } from '@hooks/useDisplay';
 import BottomSheetDialog from '../Select/BottomSheetSelect/BottomSheetDialog';
 import { fontsObject } from '@sopt-makers/fonts';
 import CalendarIcon from '@assets/svg/calendar_big.svg';
+import CalendarMobileIcon from '@assets/svg/calendar_small.svg';
+import { useFormContext } from 'react-hook-form';
+import { formatCalendarDate } from '@utils/dayjs';
+import { formatDateInput, MAX_DATE_INPUT_LENGTH, WEEKDAYS } from '@utils/date';
+
+/**
+ * CalendarInputForm
+ * @param selectedDate 선택된 날짜
+ * @param setSelectedDate 선택된 날짜 변경 함수
+ * @param selectedDateFieldName 선택된 날짜 필드 이름
+ * @param error 에러 메시지
+ * @param dateType 캘린더 타입, startDate(시작일 캘린더), endDate(마감일 캘린더), singleSelect(단일선택 캘린더)
+ */
+
 interface Props {
-  selectedDate: string | null;
-  setSelectedDate: Dispatch<SetStateAction<string | null>>;
+  selectedDate: string[] | null;
+  setSelectedDate: Dispatch<SetStateAction<string[] | string | null>>;
+  selectedDateFieldName: string;
   error?: string;
+  dateType?: 'startDate' | 'endDate' | 'singleSelect';
 }
 
-const CalendarInputForm = ({ selectedDate, setSelectedDate, error }: Props) => {
+const CalendarInputForm = ({ selectedDate, setSelectedDate, error, dateType, selectedDateFieldName }: Props) => {
   const [isOpen, setIsOpen] = useState(false);
+  const [startDate, endDate] = selectedDate ?? ['', ''];
+  const newValue = dateType === 'endDate' ? selectedDate?.[1] : selectedDate?.[0];
 
-  const CalendarComponent = () => {
-    return (
-      <>
-        <Calendar
-          value={selectedDate ? dayjs(selectedDate, 'YYYY-MM-DD').toDate() : null}
-          onClickDay={date => setSelectedDate(dayjs(date).format('YYYY.MM.DD'))}
-          formatDay={(locale, date) => dayjs(date).format('D')}
-          formatShortWeekday={(locale, date) => ['SUN', 'MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT'][date.getDay()] ?? ''}
-          showNeighboringMonth={false}
-          next2Label={null}
-          prev2Label={null}
-          minDetail="month"
-          maxDetail="month"
-          calendarType="US"
-          tileContent={({ date, view }) => {
-            if (selectedDate == dayjs(date).format('YYYY.MM.DD')) {
-              return (
-                <SDotWrapper>
-                  <SDot></SDot>
-                </SDotWrapper>
-              );
-            }
-          }}
-        />
-        {error && <SErrorMessage>{error}</SErrorMessage>}
-      </>
-    );
+  const handleDateSelection = (newDate: string) => {
+    if (dateType === 'singleSelect') {
+      setSelectedDate([newDate, '']);
+      return;
+    }
+
+    if (startDate && endDate) {
+      setSelectedDate([newDate, '']);
+      return;
+    }
+
+    if (!startDate && !endDate) {
+      setSelectedDate([newDate, '']);
+      return;
+    }
```

Comment:
```suggestion
    if ((dateType === 'singleSelect')  || (startDate && endDate) || (!startDate && !endDate)) {
      setSelectedDate([newDate, '']);
      return;
    }
```

이렇게 하면 코드 중복을 줄일 수 있겠네요!

Reply:
우왓 감사합니다 역시 코테의 신..